numpy/doc/subclassing.py

```diff
@@ -555,6 +555,46 @@ def __array_wrap__(self, arr, context=None):
 how this can work, have a look at the ``memmap`` class in
 ``numpy.core``.
 
+Subclassing and Downstream Compatibility
+----------------------------------------
+
+When sub-classing ``ndarray`` or creating duck-types that mimic the ``ndarray``
+interface, it is your responsibility to decide how aligned your APIs will be
+with those of numpy. There is sometimes great temptation and confusion amongst
+users of libraries downstream from numpy to use numpy methods on downstream
+objects that may actually not be compatible with numpy's methods.
+
+For example, if you want your sub-class or duck-type to be compatible with
+numpy's ``sum`` method, the method signature for this object's ``sum`` method
+should be the following:
+
+.. testcode::
+
+    def sum(self, axis=None, dtype=None, out=None, keepdims=False):
+    ...
+
+This is the exact same method signature for ``np.sum``, so now if a user calls
+``np.sum`` on this object, numpy will call the object's own ``sum`` method and
+pass in these arguments enumerated above in the signature, and no errors will
+be raised because the signatures are completely compatible with each other.
+
+If, however, you decide to deviate from this signature and do something like this:
+
+.. testcode::
+
+   def sum(self, axis=None, dtype=None):
+   ...
+
+This object is no longer compatible with numpy because if you call ``np.sum`` now,
+it will pass in unexpected arguments ``out`` and ``keepdims``, causing a TypeError
+to be raised.
+
+As the creator or author, if you do decide to deviate from the numpy API (e.g.
```

Comment:
I would also drop this paragraph, because it is mostly opinion.

Good documentation should be descriptive, not prescriptive.